packages/store/src/__tests__/database.unit.spec.ts

```diff
@@ -0,0 +1,279 @@
+  let col1: Column['id'];
+  let col2: Column['id'];
+  let col3: Column['id'];
+
+  const selection = [
+    { value: 'Done', color: '#F5F5F5' },
+    { value: 'TODO', color: '#FFE1E1' },
+    { value: 'WIP', color: '#E1EFFF' },
+  ];
+
+  beforeEach(() => {
+    const options = createTestOptions();
+    workspace = new Workspace(options).register(AffineSchemas);
+    page = workspace.createPage('page0');
+    page.awarenessStore.setFlag('enable_database', true);
+    db = page.db;
+
+    pageBlockId = page.addBlock('affine:page', {
+      title: new page.Text('database test'),
+    });
+    frameBlockId = page.addBlock('affine:frame', {}, pageBlockId);
+
+    col1 = page.db.updateColumn({
+      name: 'Number',
+      type: 'number',
+      width: 200,
+      hide: false,
+      decimal: 0,
+    });
+    col2 = page.db.updateColumn({
+      name: 'Single Select',
+      type: 'select',
+      width: 200,
+      hide: false,
+      selection,
+    });
+    col3 = page.db.updateColumn({
+      name: 'Rich Text',
+      type: 'rich-text',
+      width: 200,
+      hide: false,
+    });
+
+    databaseBlockId = page.addBlock(
+      'affine:database',
+      {
+        columns: [col1, col2, col3],
+        titleColumn: 'Title',
+      },
+      frameBlockId
+    );
+    p1 = page.addBlock(
+      'affine:paragraph',
+      {
+        text: new page.Text('text1'),
+      },
+      databaseBlockId
+    );
+    p2 = page.addBlock(
+      'affine:paragraph',
+      {
+        text: new page.Text('text2'),
+      },
+      databaseBlockId
+    );
+
+    page.db.updateCell(p1, {
+      columnId: col1,
+      value: 0.1,
+    });
+    page.db.updateCell(p2, {
+      columnId: col2,
+      value: [selection[1]],
+    });
+  });
+
+  test('getColumn', () => {
+    const column: Column = {
+      id: 'testColumnId',
+      name: 'Test Column',
+      type: 'number',
+      width: 100,
+      hide: false,
+    };
+    db.updateColumn(column);
+
+    const result = db.getColumn(column.id);
+    expect(result).toEqual(column);
+  });
+
+  test('updateColumn', () => {
+    const column: Omit<Column, 'id'> & { id?: Column['id'] } = {
+      name: 'Test Column',
+      type: 'number',
+      width: 100,
+      hide: false,
+    };
+
+    const id = db.updateColumn(column);
+    const result = db.getColumn(id);
+
+    expect(result).toMatchObject(column);
+    expect(result).toHaveProperty('id');
+  });
+
+  test('deleteColumn', () => {
+    const columnId = 'testColumnId';
+    const column: Column = {
+      id: columnId,
+      name: 'Test Column',
+      type: 'number',
+      width: 100,
+      hide: false,
+    };
+
+    db.updateColumn(column);
+    expect(db.getColumn(columnId)).toEqual(column);
+
+    db.deleteColumn(columnId);
+    expect(db.getColumn(columnId)).toBeNull();
+  });
+
+  test('getCell', () => {
+    const modelId = page.addBlock('affine:paragraph', {
+      text: new page.Text('paragraph'),
+    });
+    const schema: Column = {
```

Comment:
Name this as `column` will be better. I plan to remove the concept of `ColumnSchema` and make it `Column`.